rai/lib/blocks.cpp

```diff
@@ -196,29 +196,29 @@ bool rai::send_block::deserialize_json (boost::property_tree::ptree const & tree
 		auto balance_l (tree_a.get<std::string> ("balance"));
 		auto work_l (tree_a.get<std::string> ("work"));
 		auto signature_l (tree_a.get<std::string> ("signature"));
-		result = hashables.previous.decode_hex (previous_l);
-		if (!result)
+		error = hashables.previous.decode_hex (previous_l);
+		if (!error)
 		{
-			result = hashables.destination.decode_account (destination_l);
-			if (!result)
+			error = hashables.destination.decode_account (destination_l);
+			if (!error)
 			{
-				result = hashables.balance.decode_hex (balance_l);
-				if (!result)
+				error = hashables.balance.decode_hex (balance_l);
+				if (!error)
 				{
-					result = rai::from_string_hex (work_l, work);
-					if (!result)
+					error = rai::from_string_hex (work_l, work);
+					if (!error)
 					{
-						result = signature.decode_hex (signature_l);
+						error = signature.decode_hex (signature_l);
 					}
 				}
 			}
 		}
 	}
 	catch (std::runtime_error const &)
 	{
-		result = true;
+		error = true;
```

Comment:
What exception are we eating here?  failures from `tree.get<>()`?  Besides "the node at that path was not present" what other failure modes get thrown out here?  It would be great if we could record these in a log.  Or at least leave a comment describing what we expect to catch here.

Thanks for cleaning this up, but no good deed goes unpunished. ;)

Reply:
Well, you know how I feel about exceptions and eating them ;) In my opinion, making sure errors are logged and not eaten is a separate patch, this is just renaming. 

I think there's a lot of places that happens. There's also the issue of boost logger not being available everywhere (for instance, some io callbacks which now fail silently)